subsys/net/lib/mqtt_sn/mqtt_sn_transport_udp.c

```diff
@@ -63,7 +63,7 @@ static int tp_udp_init(struct mqtt_sn_transport *transport)
 	}
 
 	if (IS_ENABLED(CONFIG_MQTT_SN_LOG_LEVEL_DBG)) {
-		char ip[30], *out;
+		char ip[40], *out;
```

Comment:
Please replace the magic constant by `NET_INET6_ADDRSTRLEN` which contains the max length of the IPv6 address.